tests/tests/blob.rs

```diff
@@ -0,0 +1,148 @@
+#![allow(non_snake_case)]
+
+use fuel_core::{
+    database::{
+        database_description::on_chain::OnChain,
+        Database,
+    },
+    service::Config,
+};
+use fuel_core_bin::FuelService;
+use fuel_core_client::client::{
+    types::TransactionStatus,
+    FuelClient,
+};
+use fuel_core_types::{
+    fuel_asm::{
+        op,
+        GTFArgs,
+        RegId,
+    },
+    fuel_tx::{
+        BlobBody,
+        BlobId,
+        BlobIdExt,
+        TransactionBuilder,
+    },
+    fuel_types::canonical::Serialize,
+    fuel_vm::checked_transaction::IntoChecked,
+};
+
+struct TestContext {
+    _node: FuelService,
+    client: FuelClient,
+}
+impl TestContext {
+    async fn new() -> Self {
+        let config = Config {
+            debug: true,
+            utxo_validation: false,
+            ..Config::local_node()
+        };
+
+        let _node = FuelService::from_database(Database::<OnChain>::in_memory(), config)
+            .await
+            .unwrap();
+        let client = FuelClient::from(_node.bound_address);
+
+        Self { _node, client }
+    }
+
+    async fn new_blob(&mut self, blob_data: Vec<u8>) -> (TransactionStatus, BlobId) {
+        let blob_id = BlobId::compute(&blob_data);
+        let tx = TransactionBuilder::blob(BlobBody {
+            id: blob_id,
+            witness_index: 0,
+        })
+        .add_witness(blob_data.into())
+        .add_random_fee_input()
+        .finalize_as_transaction()
+        .into_checked(Default::default(), &Default::default())
+        .expect("Cannot check transaction");
+
+        let status = self
+            .client
+            .submit_and_await_commit(tx.transaction())
+            .await
+            .expect("Cannot submit blob");
+        (status, blob_id)
+    }
+}
+
+#[tokio::test]
+async fn blob__upload_works() {
+    // Given
+    let mut ctx = TestContext::new().await;
+    let (status, blob_id) = ctx
+        .new_blob([op::ret(RegId::ONE)].into_iter().collect())
+        .await;
+    assert!(matches!(status, TransactionStatus::Success { .. }));
+
+    // When
+    let script_tx = TransactionBuilder::script(
```

Comment:
Instead of using `new_blob` shouldn't the "when" of this test be sending the `Blob` tx?

The `bsiz` is just how you chose to verify that the blob is callable, right?

Reply:
Yes, and yes. Fixed in a0bdfcc43.